src/lasp_plumtree_broadcast_distribution_backend.erl

```diff
@@ -261,24 +261,35 @@ query(Id) ->
 %%
 -spec update(id(), operation(), actor()) -> {ok, var()} | error().
 update(Id, Operation, Actor) ->
-    {ok, {Id, Type, Metadata, Value}} = gen_server:call(?MODULE,
-                                                        {update, Id, Operation, Actor},
-                                                        infinity),
-    ReturnState = case orddict:find(dynamic, Metadata) of
-                      {ok, true} ->
-                          %% Ignore: this is a dynamic variable.
-                          Value;
-                      _ ->
-                          case Value of
-                              {delta, MergedState} ->
-                                  %% No broadcasting for the delta.
-                                  MergedState;
+    case gen_server:call(?MODULE, {update, Id, Operation, Actor}, infinity) of
+        {ok, {Id, Type, Metadata, Value}} ->
+            ReturnState = case orddict:find(dynamic, Metadata) of
+                              {ok, true} ->
+                                  %% Ignore: this is a dynamic variable.
+                                  Value;
                               _ ->
-                                  broadcast({Id, Type, Metadata, Value}),
-                                  Value
-                          end
-                  end,
-    {ok, {Id, Type, Metadata, ReturnState}}.
+                                  case Value of
+                                      {delta, MergedState} ->
+                                          %% No broadcasting for the delta.
+                                          MergedState;
+                                      _ ->
+                                          broadcast({Id, Type, Metadata, Value}),
+                                          Value
+                                  end
+                          end,
+            {ok, {Id, Type, Metadata, ReturnState}};
+        {error, not_found} ->
+            case Id of
+                {StorageId, TypeId} ->
+                    declare(StorageId, TypeId),
+                    gen_server:call(?MODULE, {update, Id, Operation, Actor}, infinity);
```

Comment:
That appears to be a race condition on shutting down -- something is calling the bind function after the lasp_plumtree_broadcast_distribution_backend process has terminated.